### PR TITLE
[IoTCentral] new version of swagger, modify error definition, and remove legacy skus

### PR DESCRIPTION
**Pull request by kawayiboy**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/kawayiboy/azure-rest-api-specs/telu%2Ferrdefversion...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/examples/Apps_CheckNameAvailability.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "parameters": {
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "api-version": "2018-09-01",
```
Comment (MyronFanQiu) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/examples/Apps_CheckNameAvailability.json]: Hello. Looks like this PR doesn't change the api-version. Does it work in progress?

**Review by mmyyrroonn — CHANGES_REQUESTED** [review]
If this PR is in progress, please add WIP into the title.

File: specification/iotcentral/resource-manager/readme.ruby.md [hunk 2]
```diff
@@ -12,10 +12,21 @@ azure-arm: true
 
 ``` yaml $(ruby) && $(multiapi)
 batch:
+  - tag: package-2020-03-01
```
Comment (r-delgadillo) [inline specification/iotcentral/resource-manager/readme.ruby.md]: Is this API change required since it's just an SDK breaking change and not an RP

**Review by r-delgadillo — CHANGES_REQUESTED** [review]
Let's discuss with the team to see if there's any additional changes we can include in this breaking change.

File: specification/iotcentral/resource-manager/readme.ruby.md (hunk 2 above)
Comment (kawayiboy) [reply specification/iotcentral/resource-manager/readme.ruby.md]: ok

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/examples/Apps_CheckNameAvailability.json (hunk 1 above)
Comment (kawayiboy) [reply specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/examples/Apps_CheckNameAvailability.json]: fixed

**Comment by MyronFanQiu** [comment]
@kawayiboy Hi. Is this PR ready for review?

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/iotcentral.json [hunk 3]
```diff
@@ -0,0 +1,868 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-09-01",
```
Comment (MyronFanQiu) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/iotcentral.json]: Hello. Looks like this was not fixed.

File: specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/iotcentral.json [hunk 4]
```diff
@@ -0,0 +1,868 @@
...
+            }
+          },
+          "default": {
+            "description": "Default error response",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "deprecated": false
```
Comment (MyronFanQiu) [inline specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2020-03-01/iotcentral.json]: the default value should be false. well. This is correct but not commonly appear. 😄

**Review by mmyyrroonn — CHANGES_REQUESTED** [review]
LGTM in general. Just a little comments

**Pull request closed without merging**